### ISSUE TITLE
Problems validating objects inside an array

### ISSUE THREAD

**Issue by agabi10** [express-validator/express-validator#1184]
Hi, I'm trying to check if the data I receive in my input is an array of objects with specific properties. It's something like this:

```json
[
    { "character": "A", "conversion-type": 1, "message": "asdf" },
    { "character": "B", "conversion-type": 2, "message": "dfgh" }
]
```

In this example the conditions that the message property must meet are different depending on the value of the conversion-type property, so the validation chain I did is something like this:

```js
[
    body().isArray().withMessage('The body of the request should be an array'),
    body('*').isObject().bail().custom((value) => {
        if ('character' in value === false || typeof value.character !== string || !value.character.match(/^[A-Za-z]+$/))
            throw new Error('Invalid value for character');

       if ('conversion-type' in value === false || !validConversions.includes(value['conversion-type']))
           throw new Error('Invalid value for conversion-type');

       switch(value['conversion-type']) {
            // check each of the possible conversion types and throw an error if the message doesn't 
            // have the expected format for that conversion type
       }
    })
]
```

The problems I have with this is that the second validation checks all the elements regardless of the first one failing, so if the client doesn'd use an array for the request there are lots of errors telling me that all of the properties of the object have an invalid format and I'd like to avoid sending those errors in the response.

The other problem I have is that all of the objects in the array should have some mandatory properties and the values of some of the properties depend on the values that other properties have, and the only way I found to make that is having a custom validator. I tried using wildcards like in `'*.character'`, `'*.conversion-type'` and `'*.message'`, but I didn't find a way to check the conversion type in the custom validator of the message property, and in the case the body provided being an object instead of an array there are even more error messages when the only error I'd like to show in the response is that an array of elements was expected. 

Is there something I can do to achieve what I want?

**Comment by gustavohenke** [comment]
Hey!

> problems I have with this is that the second validation checks all the elements regardless of the first one failing

Good news, this is being implemented for v7.0.0 🙂 it'll be called **request-level bail**.

> I tried using wildcards like in '*.character', '*.conversion-type' and '*.message', but I didn't find a way to check the conversion type in the custom validator of the message property

I'm not sure if I understand this. Can you please clarify it?

**Comment by gustavohenke** [comment]
Closing due to inactivity.

Request-level bail is implemented, if that's of interest.
https://express-validator.github.io/docs/api/validation-chain/#bail
